lib/standard/string.nit

```diff
@@ -864,6 +864,23 @@ abstract class FlatText
 	# Real items, used as cache for to_cstring is called
 	private var real_items: nullable NativeString = null
 
+	# Returns a char* starting at position `index_from`
+	#
+	# WARNING: If you choose to use this service, be careful of the following.
+	#
+	# Strings and NativeString are *ideally* always allocated through a Garbage Collector.
+	# Since the GC tracks the use of the pointer for the beginning of the char*, it may be
+	# deallocated at any moment, rendering the pointer returned by this function invalid.
+	# Any access to freed memory may very likely cause undefined behaviour or a crash.
+	# (Failure to do so will most certainly result in long and painful debugging hours)
+	#
+	# The only safe use of this pointer is if it is ephemeral (e.g. read in a C function
+	# then immediately return).
+	#
+	# As always, do not modify the content of the String in C code, if this is what you want
+	# copy locally the char* as Nit Strings are immutable.
+	private fun fast_cstring: NativeString is abstract
```

Comment:
That's way better!

You insist a bit too much on what happens when using deallocated memory, it should tell us what to do right instead. Here's what I propose, merging the "undefined behavior" part to the next paragraph:

	# ...rendering the pointer returned by this function invalid.
	#
	# In foreign code, do not store the returned reference and do not callback to Nit when using it, this may trigger the GC and cause undefined behavior.
	#
	# As always, do not modify the content of the String in C code, ...

This is a standard problem with optimized FFIs, and the 2 "don't" should prevent all errors.

Last detail, there is some inconsistency between using NativeString, char * and String in C for the same type. I like `char*` but I may have a bias...

Oops, one very last thing, maybe "the Garbage Collector" instead of "a Garbage Collector"? Because in my programs we often have the JVM's GC which tries to do its thing.